File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, nums: List[int]) -> int:
        n = len(nums)
        mn, mx, x = 101, -1, 0
        ans = 0

        while x < n:
            if nums[x] < mn:
                mn = nums[x]
                mx = nums[x]
            
            if nums[x] > mx:
                mx = nums[x]
                ans = max(ans, mx-mn)


            x += 1
        
        return ans

        

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        int l = 0, h = 1;

        while(l < n && h < n) {
            if(nums[l] != nums[h]){
                nums[l+1] = nums[h];
                l++;
            }
            h++;
        }

        return l+1;
    }
};